pennylane/ops/qubit/matrix_ops.py

```diff
@@ -218,7 +218,7 @@ def compute_sparse_matrix(U: TensorLike):  # pylint: disable=arguments-differ
             with 2 stored elements in Compressed Sparse Row format>
         """
         if sp.sparse.issparse(U):
-            return U.tocsr()
+            return U.tocsr().asformat(format)
```

Comment:
```suggestion
            return U.asformat(format)
```
I guess we don't need to `tocsr` any more.